### PR TITLE
ServiceNow v2 - add sysparm_display_value to update/create commands

### PR DESCRIPTION
**Pull request by bakatzir**
- Added the ***input_display_value*** to the *servicenow-update-ticket*, *servicenow-update-record* commands.

## Status
- [x] In Progress - customer verification

## Related Issues
fixes: https://github.com/demisto/etc/issues/26879

## Minimum version of Demisto
- [x] 5.0.0

## Does it break backward compatibility?
   - [x] No

- [x] Docs

### CODE REVIEW

File: Packs/ServiceNow/Integrations/ServiceNowv2/ServiceNowv2.py [hunk 1]
```diff
@@ -839,11 +841,11 @@ def update_ticket_command(client: Client, args: dict) -> Tuple[Any, Dict, Dict,
     ticket_id = str(args.get('id', ''))
     additional_fields = split_fields(str(args.get('additional_fields', '')))
     additional_fields_keys = list(additional_fields.keys())
-
     fields = get_ticket_fields(args, ticket_type=ticket_type)
     fields.update(additional_fields)
+    input_display_value = args.get('fields', 'input_display_value') == 'true'
```
Comment (liorblob) [inline Packs/ServiceNow/Integrations/ServiceNowv2/ServiceNowv2.py:846]: Use `str2bool`

File: Packs/ServiceNow/Integrations/ServiceNowv2/ServiceNowv2.py [hunk 2]
```diff
@@ -1251,6 +1253,7 @@ def update_record_command(client: Client, args: dict) -> Tuple[Any, Dict[Any, An
     record_id = str(args.get('id', ''))
     fields_str = str(args.get('fields', ''))
     custom_fields_str = str(args.get('custom_fields', ''))
+    input_display_value = args.get('fields', 'input_display_value') == 'true'
```
Comment (liorblob) [inline Packs/ServiceNow/Integrations/ServiceNowv2/ServiceNowv2.py:1256]: Use `str2bool`

File: Packs/ServiceNow/Integrations/ServiceNowv2/ServiceNowv2.yml [hunk 3]
```diff
@@ -1248,6 +1249,18 @@ script:
       name: additional_fields
       required: false
       secret: false
+    - auto: PREDEFINED
+      default: false
+      defaultValue: 'false'
+      description: Flag that indicates whether to set field values using the display
```
Comment (liorblob) [inline Packs/ServiceNow/Integrations/ServiceNowv2/ServiceNowv2.yml:1255]: I didn't understand this. Maybe an example can help.

File: Packs/ServiceNow/Integrations/ServiceNowv2/ServiceNowv2.yml [hunk 4]
```diff
@@ -1686,6 +1699,18 @@ script:
       name: custom_fields
       required: false
       secret: false
+    - auto: PREDEFINED
+      default: false
+      defaultValue: 'false'
+      description: Flag that indicates whether to set field values using the display
```
Comment (liorblob) [inline Packs/ServiceNow/Integrations/ServiceNowv2/ServiceNowv2.yml:1705]: same here

File: Packs/ServiceNow/Integrations/ServiceNowv2/ServiceNowv2.py (hunk 1 above)
Comment (bakatzir) [reply Packs/ServiceNow/Integrations/ServiceNowv2/ServiceNowv2.py:846]: str2bool is in the sdk, used `argToBoolean`

File: Packs/ServiceNow/Integrations/ServiceNowv2/ServiceNowv2.py (hunk 2 above)
Comment (bakatzir) [reply Packs/ServiceNow/Integrations/ServiceNowv2/ServiceNowv2.py:1256]: str2bool is in the sdk, used `argToBoolean`

File: Packs/ServiceNow/Integrations/ServiceNowv2/ServiceNowv2.yml (hunk 3 above)
Comment (bakatzir) [reply Packs/ServiceNow/Integrations/ServiceNowv2/ServiceNowv2.yml:1255]: added desc

File: Packs/ServiceNow/Integrations/ServiceNowv2/ServiceNowv2.yml (hunk 4 above)
Comment (bakatzir) [reply Packs/ServiceNow/Integrations/ServiceNowv2/ServiceNowv2.yml:1706]: added desc

Comment (liorblob) [reply Packs/ServiceNow/Integrations/ServiceNowv2/ServiceNowv2.yml:1706]: Can you add an example?

Comment (bakatzir) [reply Packs/ServiceNow/Integrations/ServiceNowv2/ServiceNowv2.yml:1706]: Added to readme

**Comment by bakatzir** [comment]
Customer verified the solution.

File: Packs/ServiceNow/Integrations/ServiceNowv2/README.md [hunk 5]
```diff
@@ -374,7 +374,9 @@ Updates the specified ticket.
 | caller | Caller system ID. | Optional | 
 | approval | Ticket approval. | Optional | 
 | additional_fields | Additional fields in the format: fieldname1=value;fieldname2=value; | Optional | 
+| input_display_value | Flag that indicates whether to set field values using the display value or the actual value. true will treat input value as the dispaly value. false treats input values as actual values. Default is false. | Optional |
```
Comment (amshamah419) [inline Packs/ServiceNow/Integrations/ServiceNowv2/README.md:377]:
```

| input_display_value | Flag that indicates whether to set field values using the display value or the actual value. True will treat the input value as the display value. False treats the input values as actual values. The default setting is false. | Optional |

```

File: Packs/ServiceNow/Integrations/ServiceNowv2/README.md [hunk 6]
```diff
@@ -902,8 +904,10 @@ Updates a record in the specified ServiceNow table.
 | table_name | The name of the table to update the record in. | Required | 
 | id | The system ID of the ticket to update. | Required | 
 | fields | Fields and their values to update in the record, in the format: fieldname1=value;fieldname2=value;... | Optional | 
-| custom_fields | Custom (user defined) fields and their values to update in the record, in the format: fieldname1=value;fieldname2=value;... | Optional | 
+| custom_fields | Custom (user defined) fields and their values to update in the record, in the format: fieldname1=value;fieldname2=value;... | Optional |
+| input_display_value | Flag that indicates whether to set field values using the display value or the actual value. true will treat input value as the dispaly value. false treats input values as actual values. Default is false. | Optional |
```
Comment (amshamah419) [inline Packs/ServiceNow/Integrations/ServiceNowv2/README.md:908]:
```
| input_display_value | Flag that indicates whether to set field values using the display value or the actual value. True will treat the input value as the display value. False treats the input values as actual values. The default setting is false. | Optional |

```

**Comment by bakatzir** [comment]
> cool feature.
> consider adding UT that just checks that the sysparm_input_display_value is sent correctly to prevent a break it in the future.

@yuvalbenshalom A check for that is in the `argToBoolean` that handles it.
We cannot check the local vars in the code, and do not get an indication from SNOW that this param was sent.
I can add import the `argToBoolean` to SNOW_test and validate it there as well, but maybe i'm missing your intention.

**Pull request merged**